cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -475,4 +480,17 @@ private NamespaceId getNamespaceId(byte[] rowKey) {
   public interface DeleteCallback {
     void handle(NamespaceId namespaceId, Location location, String namespacedLogBaseDir);
   }
+
+  private static Location getCompatibleLocationFromValue(LocationFactory locationFactory,
+                                                         byte[] value) throws URISyntaxException {
```

Comment:
Based on the discussion, rootLocationFactory would be used to create the log files, so it can be used from inside the method. Making non-static.